gammapy/cube/basic_cube.py

```diff
@@ -0,0 +1,62 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Functions to perform basic functions for map and cube analysis.
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+from ..maps import WcsNDMap
+
+def make_map_counts(evts, valid_ref_map):
+    """
+    Build a ``WcsNDMap` with events from an EventList.
+    The energy of the events is used for a non-spatial axis homogeneous to energy.
+    The other non-spatial axis names should have an entry in the colum names of the ``EventList``
+
+    Parameters
+    __________
+
+    evts : `~gammapy.data.EventList`
+            the input event list
+    valid_ref_map : `~gammapy.maps.WcsNDMap`
+        Map containing the valid pixels
+
+    Returns
+    -------
+    cntmap : `~gammapy.maps.WcsNDMap`
+        Count cube (3D) in true energy bins
+
+    """
+    # The list will contain the event table entries to be fed into the WcsNDMap
+    tmp = list()
+    ref_geom = valid_ref_map.geom
+    # Convert events coordinates
+    if ref_geom.coordsys == 'GAL':
+        tmp.append(evts.galactic.l)
+        tmp.append(evts.galactic.b)
+    elif ref_geom.coordsys == 'CEL':
+        tmp.append(evts.radec.ra)
+        tmp.append(evts.radec.dec)
+    else:
+        # should raise an error here. The map is not correct.
+        raise ValueError("Incorrect coordsys of input map.")
+
+    for i, axis in enumerate(ref_geom.axes):
+        if axis.unit.is_equivalent("eV"):
```

Comment:
we should have a declarative scheme, via a "name" or "type" for energy, not guess via a unit. Not sure if it's possible at the moemnt.

Reply:
If we impose that ``energy_reco`` or ``energy`` are required names, we can.

Reply:
@woodmd - how do you feel about introducing a "type" attribute for
http://docs.gammapy.org/dev/api/gammapy.maps.MapAxis.html
and to have the following options:
- lon - a longitude
- lat - a latitude
- energy - an energy
- time - a time
- any - anything else

and then generic code like here uses that (and equivalent special propoerties on events or IRFs).

I think checking for unit equivalency is ugly.

So either introduce "axis type" with a few special cases, or use names. But using names is problematic, becuase of the different energies, different frames glon / glat vs ra / dec, ..., no?

Reply:
I agree that having a "type" attribute would be cleaner than checking the unit.  In some ways this would serve a similar role to UCD in FITS.  I'd be fine with adding this to `MapAxis`.

Reply:
I agree also that it will be better to introduce this "axis type" in maps!

Reply:
In HESS we have this concept in the IRF software also. There are special axes and their type also controls interpolation / extrapolation behaviour (e.g. phi should ideally wrap and do circular interpolation).